book/src/design/circuit/gadgets/ecc/fixed-base-scalar-mul.md

```diff
@@ -0,0 +1,83 @@
+# Fixed-base scalar multiplication
+
+There are $6$ fixed bases in the Orchard protocol:
+- $\mathcal{K}^{\mathsf{Orchard}}$, used in deriving the nullifier;
+- $\mathcal{G}^{\mathsf{Orchard}}$, used in spend authorization;
+- $\mathcal{R}$ base for $\mathsf{NoteCommit}^{\mathsf{Orchard}}$;
+- $\mathcal{V}$ and $\mathcal{R}$ bases for $\mathsf{ValueCommit}^{\mathsf{Orchard}}$; and
+- $\mathcal{R}$ base for $\mathsf{Commit}^{\mathsf{ivk}}$.
+
+## Witness scalar
+In most cases, we multiply the fixed bases by $255-$bit scalars from $\mathbb{F}_q$. We decompose a full-width scalar $\alpha$ into $85$ $3$-bit windows:
+
+$$\alpha = k_0 + k_1 \cdot (2^3)^1 + \cdots + k_{84} \cdot (2^3)^{84}, k_i \in [0..2^3).$$
```

Comment:
This doesn't seem correct to me. \alpha actually does have to be canonical, because if it's only correct modulo p, that is different from it being correct modulo q (i.e. in the scalar field).

Reply:
My point is that we never need to represent `alpha` as a base field element, or compute with it modulo p; it is witnessed purely as bits, and the fixed-base scalar mul will calculate the correct result whether or not `alpha` is canonical modulo q. If you are assuming something different, please say so!

Reply:
If necessary, my suggestion can be clarified thusly:
```
$\alpha$ is allowed to be non-canonical modulo $q$; ...
```

Reply:
I think saying "$\alpha$ is allowed to be non-canonical" does more harm than good. We do need the scalar used in nullifier computation to be canonical, otherwise there would be a double-spend bug! How about this:

```suggestion
$$\alpha = k_0 + k_1 \cdot (2^3)^1 + \cdots + k_{84} \cdot (2^3)^{84}, k_i \in [0..2^3).$$

The scalar multiplication will be computed correctly for $k_{0..84}$ representing any integer in the range $[0, 2^{255})$. If $k_{0..84}$ is witnessed directly then no issue of canonicity arises. If the scalar is given as a base field element, then care must be taken to ensure a canonical representation, since $2^{255} > p$. This occurs, for example, in the scalar multiplication for the nullifier computation of the Action circuit.

$$
\begin{array}{|c|l|}
\hline
\text{Degree} & \text{Constraint} \\\hline
9 & q_\text{scalar-fixed} \cdot 1 \cdot \left(\sum\limits_{i=0}^7{w - i}\right) = 0 \\\hline
\end{array}
$$
```